openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/runtime/AzureOpenAiRecorder.java

```diff
@@ -85,18 +92,44 @@ public Supplier<?> embeddingModel(Langchain4jAzureOpenAiConfig runtimeConfig) {
 
         return new Supplier<>() {
             @Override
-            public Object get() {
+            public EmbeddingModel get() {
                 return builder.build();
             }
         };
     }
 
-    private String getBaseUrl(Langchain4jAzureOpenAiConfig runtimeConfig) {
-        var baseUrl = runtimeConfig.baseUrl();
+    static String getEndpoint(Langchain4jAzureOpenAiConfig runtimeConfig) {
```

Comment:
You know what I'm going to say: 

No lambdas in runtime code please

Reply:
I'll make the changes on Monday, but.....

One of these days I'd like to have you really explain to me the significance of lambdas in runtime code. Not that I don't believe you, but because I'd like to understand.

The code as written is so much cleaner and easier to read. `Optional.isPresent()` followed by `get()` is a total anti-pattern as well.

Reply:
We've had this discussion before 🙂